src/main/java/com/bandwidth/sdk/messaging/MessagingClient.java

```diff
@@ -76,4 +87,109 @@ public Message sendMessage(SendMessageRequest request) throws ExecutionException
         }
     }
 
+    /**
+     * Uploads MMS media content. Max 65MB
+     *
+     * @param path Path to file to upload
+     * @param fileName File name as you would like it to be uploaded
+     * @return URL that can be sent in an MMS
+     */
+    public String uploadMedia(String path, String fileName) throws FileNotFoundException, IOException {
+        FileInputStream stream = new FileInputStream(path);
+        return uploadMedia(stream, fileName);
+    }
+
+    /**
+     * Uploads MMS media content. Max 65MB
+     *
+     * @param stream InputStream that contains the data to be sent
+     * @param fileName File name as you would like it to be uploaded
+     * @return URL that can be sent in an MMS
+     */
+    public String uploadMedia(InputStream stream, String fileName) throws IOException {
+        byte[] byteArray = ByteStreams.toByteArray(stream);
+        return uploadMedia(byteArray, fileName);
+    }
+
+    /**
+     * Uploads MMS media content. Max 65MB
+     *
+     * @param byteArray byte array of the data to be sent
+     * @param fileName File name as you would like it to be uploaded
+     * @return URL that can be sent in an MMS
+     */
+    public String uploadMedia(byte[] byteArray, String fileName) {
+        return uploadMediaAsync(byteArray, fileName).join();
+    }
+
+    /**
+     * Uploads MMS media content. Max 65MB
+     *
+     * @param byteArray byte array of the data to be sent
+     * @param fileName File name as you would like it to be uploaded
+     * @return CompletableFuture that contains URL that can be sent in an MMS
+     */
+    public CompletableFuture<String> uploadMediaAsync(byte[] byteArray, String fileName){
+        String url = MessageFormat.format("{0}/users/{1}/media", MEDIA_URL, userId);
+        try{
+            return httpClient.preparePut(url)
+                    .setBody(byteArray)
+                    .execute()
+                    .toCompletableFuture()
+                    .thenApply((resp) -> {
+                        if (resp.getStatusCode() != 200)
```

Comment:
Same as above, check the 2xx range. (Maybe move to a separate function too since it's used multiple times)